Description: Replace irq_linear_revmap with irq_find_mapping for newer kernel
 irq_linear_revmap() was removed in newer kernels (6.x+); 
 use irq_find_mapping() instead.
Author: Canonical Workshop

--- a/z9332f/modules/cls-i2c-mux-pca954x.c
+++ b/z9332f/modules/cls-i2c-mux-pca954x.c
@@ -329,7 +329,7 @@
 
 	pending = (ret >> PCA954X_IRQ_OFFSET) & (BIT(data->chip->nchans) - 1);
 	for_each_set_bit(i, &pending, data->chip->nchans)
-		handle_nested_irq(irq_linear_revmap(data->irq, i));
+		handle_nested_irq(irq_find_mapping(data->irq, i));
 
 	return IRQ_RETVAL(pending);
 }
